items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -70,10 +119,27 @@ class SierraItemUpdater(sierraService: SierraService)(
         SierraItemIdentifier.fromSourceIdentifier(srcId) -> item
     } toMap
 
-    sierraService
-      .getAccessConditions(itemMap.keys.toSeq)
-      .map(
-        updateAccessConditions(itemMap, _)
-      )
+    val accessConditions = for {
+      accessConditionsMap <- getAccessConditions(itemMap.keys.toSeq)
+
+      // It is possible for there to be a situation where Sierra does not know about
+      // an Item that is in the Catalogue API (e.g. it's been deleted but the change
+      // has not yet propagated. In that case it gets status "NotRequestable")
```

Comment:
```suggestion
      // has not yet propagated. In that case it gets method "NotRequestable")
```

I would add here that this should be extremely rare in practice – items that are removed are usually suppressed in Sierra, then deleted after a long delay. (Do we handle suppressed items correctly?)

We should log a warning here so we can see if it's happening a lot.